airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -0,0 +1,452 @@
+
+        start_point = self._start_date
+        if stream_state.get(self.cursor_field):
+            start_point = max(start_point, stream_state[self.cursor_field])
+        params["since"] = start_point
+
+        return params
+
+
+# Below are full refresh streams
+
+
+class Assignees(GithubStream):
+    pass
+
+
+class Reviews(GithubStream):
+    object_fields = ("user",)
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        pull_request_number = stream_slice["pull_request_number"]
+        return f"repos/{self.repository}/pulls/{pull_request_number}/reviews"
+
+    def stream_slices(
+        self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        pull_requests_stream = PullRequests(authenticator=self.authenticator, repository=self.repository)
+        for pull_request in pull_requests_stream.read_records(sync_mode=SyncMode.full_refresh):
+            yield {"pull_request_number": pull_request["number"]}
+
+
+class Collaborators(GithubStream):
+    pass
+
+
+class IssueLabels(GithubStream):
+    def path(self, **kwargs) -> str:
+        return f"repos/{self.repository}/labels"
+
+
+class Teams(GithubStream):
+    def path(self, **kwargs) -> str:
+        owner, _ = self.repository.split("/")
+        return f"orgs/{owner}/teams"
+
+
+# Below are semi incremental streams
+
+
+class Releases(SemiIncrementalGithubStream):
+    cursor_field = "created_at"
+    object_fields = ("author",)
+
+    def transform(self, record: Mapping[str, Any]) -> Mapping[str, Any]:
+        super().transform(record=record)
+
+        assets = record.get("assets", [])
+        for asset in assets:
+            uploader = asset.pop("uploader", None)
+            asset["uploader_id"] = uploader.get("id") if uploader else None
+
+
+class Events(SemiIncrementalGithubStream):
+    cursor_field = "created_at"
+    object_fields = (
+        "actor",
+        "repo",
+        "org",
+    )
+
+
+class PullRequests(SemiIncrementalGithubStream):
+    should_break_if_sorter = True
+    object_fields = (
+        "user",
+        "milestone",
+        "assignee",
+    )
+    list_fields = (
+        "labels",
+        "assignees",
+        "requested_reviewers",
+        "requested_teams",
+    )
+
+    def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
+        with vcr.use_cassette(cache_file.name, record_mode="new_episodes", serializer="json"):
+            yield from super().read_records(**kwargs)
+
+    def path(self, **kwargs) -> str:
+        return f"repos/{self.repository}/pulls"
+
+    def request_params(self, **kwargs) -> MutableMapping[str, Any]:
+        params = super().request_params(**kwargs)
+        params["state"] = "all"
```

Comment:
You redefine the method several times in different classes to add hard-coded values, perhaps it would be better to create a variable of the `base_params` class and add the initially hard-coded values there? I did a similar thing in Gitlab, you can see.

Reply:
Done.